packages/schema/lib/schemas/BaseFieldSchema.js

```diff
@@ -0,0 +1,210 @@
+'use strict';
+
+const makeSchema = require('../utils/makeSchema');
```

Comment:
This is a copy of `FieldSchema.js` the spit fields inherit from, I did this to decouple the `FieldSchema` incase we need to make changes to the constraints without breaking existing behavior.

Reply:
Do you mean functional constraints in lib/functional-constraints? I think, as long as there are no breaking changes to the FieldSchema, any changes/additions to the constraints shouldn't block us from inheriting directly from it. Please, let's start with inheriting directly from it for the new schemas; AuthFieldSchema, InputFieldSchema, and OutputFieldSchema.

Reply:
Will do!